api/src/main/java/org/eclipse/microprofile/faulttolerance/Asynchronous.java

```diff
@@ -29,16 +29,66 @@
 
 /**
  * Wrap the execution and invoke it asynchronously.
- * Any methods marked with this annotation must return {@link java.util.concurrent.Future}. 
+ * Any methods marked with this annotation must return one of:
+ * <ul>
+ *   <li>{@link java.util.concurrent.Future}</li>
+ *   <li>{@link java.util.concurrent.CompletionStage}</li>
+ * </ul>
  * Otherwise, {@link org.eclipse.microprofile.faulttolerance.exceptions.FaultToleranceDefinitionException} occurs.
```

Comment:
This exception should be thrown at deploy time if the class is discovered during deployment. We need to update the wording and also cover it by a test if not already.